delta/rdf/src/main/scala/ch/epfl/bluebrain/nexus/delta/rdf/jsonld/api/JsonLdJavaApi.scala

```diff
@@ -0,0 +1,150 @@
+package ch.epfl.bluebrain.nexus.delta.rdf.jsonld.api
+import akka.http.scaladsl.model.Uri
+import cats.implicits._
+import ch.epfl.bluebrain.nexus.delta.rdf.implicits._
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.IOErrorOr
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.JsonLdError._
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.JsonLdContext.keywords
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context._
+import com.github.jsonldjava.core.{Context, DocumentLoader, JsonLdProcessor, JsonLdOptions => JsonLdJavaOptions}
+import com.github.jsonldjava.utils.JsonUtils
+import io.circe.{parser, Json}
+import monix.bio.IO
+import org.apache.jena.iri.IRI
+import org.apache.jena.rdf.model.{Model, ModelFactory}
+import org.apache.jena.riot.system.StreamRDFLib
+import org.apache.jena.riot.{Lang, RDFParser, RDFWriter}
+
+import scala.jdk.CollectionConverters._
+import scala.util.Try
+
+/**
+  * Json-LD high level API implementation by Json-LD Java library
+  */
+object JsonLdJavaApi extends JsonLdApi {
+
+  System.setProperty(DocumentLoader.DISALLOW_REMOTE_CONTEXT_LOADING, "true")
+
+  // $COVERAGE-OFF$
+  override def compact[Ctx <: JsonLdContext](input: Json, ctx: Json, f: ContextFields[Ctx])(implicit
+      opts: JsonLdOptions,
+      resolution: RemoteContextResolution
+  ): IOErrorOr[(Json, Ctx)] =
+    for {
+      obj           <- catchTryErr(JsonUtils.fromString(input.noSpaces), "building input")
+      ctxObj        <- catchTryErr(JsonUtils.fromString(ctx.noSpaces), "building context")
+      options       <- remoteContextLoader(input, ctx).map(toOpts)
+      compacted     <- catchTryErr(JsonUtils.toString(JsonLdProcessor.compact(obj, ctxObj, options)), "compacting")
```

Comment:
We only want to manipulate json so maybe we could parse it in the same pass

Reply:
Every step issues different errors, so I find it cleaner as is.